### PR TITLE
Enable writing without a query

### PR DESCRIPTION
**Pull request by stubailo**

### CODE REVIEW

File: packages/apollo-link-state/src/utils.ts [hunk 1]
```diff
@@ -30,3 +39,113 @@ export function removeClientSetsFromDocument(
...
+}
+
+export function addWriteDataToCache(cache: ApolloCacheClient) {
+  cache.writeData = ({ id, data }: WriteDataArgs) => {
+    if (id) {
+      cache.writeFragment({
+        fragment: fragmentFromPojo(data),
+
+        // Add a type here to satisfy the inmemory cache
+        data: { ...data, __typename: '__FakeType' },
```
Comment (peggyrayzis) [inline packages/apollo-link-state/src/utils.ts]: Is this related to the Apollo Client errors you sent me in Slack?

File: packages/apollo-link-state/src/utils.ts [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+  FragmentDefinitionNode,
+} from 'graphql';
 
 import {
   // getDirectives,
   checkDocument,
   removeDirectivesFromDocument,
 } from 'apollo-utilities';
 
+import { ApolloCacheClient, WriteDataArgs } from './index';
```
Comment (peggyrayzis) [inline packages/apollo-link-state/src/utils.ts]: You don't need to specify index here.

File: packages/apollo-link-state/src/__tests__/noQuery.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      `;
+
+      const local = withClientState({
+        Mutation: {
+          start: (_, $, { cache }: { cache: ApolloCacheClient }) => {
+            cache.writeQuery({
+              query,
+              data: { obj: { field: 1, id: 'uniqueId', __typename: 'Object' } },
+            });
+            cache.writeData({ id: 'Object:uniqueId', data: { field: 2 } });
```
Comment (peggyrayzis) [inline packages/apollo-link-state/src/__tests__/noQuery.ts]: So cool 😍

File: packages/apollo-link-state/src/utils.ts (hunk 1 above)
Comment (stubailo) [reply packages/apollo-link-state/src/utils.ts]: Yeah if we don't have this, Apollo complians it can't resolve the fragment.

**Pull request merged**